code/modules/awaymissions/maploader/reader.dm

```diff
@@ -321,7 +325,7 @@ var/global/dmm_suite/preloader/_preloader = null
 /dmm_suite/preloader/New(list/the_attributes, path)
 	.=..()
 	if(!the_attributes.len)
-		Del()
+		qdel(src)

```

Comment:
Will that null the reference to ```preloader``` ? If not, then it will mess the whole maploader system.
(same with the other below)

Did you successfully test that change ?